WordPressKit/WordPressOrgRestApi.swift

```diff
@@ -203,8 +203,8 @@ public final class WordPressOrgRestApi: NSObject {
 private extension WordPressOrgRestApi {
     func apiBaseURL() -> URL {
         switch site {
-        case .dotCom:
-            return URL(string: "https://public-api.wordpress.com")!
+        case let .dotCom(_, _, apiURL):
+            return apiURL
```

Comment:
A nice side effect is that now we are one step closer to having a single source of truth for what this URL should be.

<img width="2550" alt="image" src="https://github.com/wordpress-mobile/WordPressKit-iOS/assets/1218433/c64ddd22-ae94-4fcd-87fa-8dfab1827657">